lib/runners/shell.rb

```diff
@@ -115,8 +115,13 @@ def capture3_trace(command, *args, **options)
         trace_writer.stderr stderr_str if trace_stderr
 
         if status.exited?
+          # normal case
           trace_writer.status status if trace_command_line
+        elsif status.signaled? && status.termsig == 12
```

Comment:
It might be better to use [`Signal.list`](https://ruby-doc.org/core-2.7.2/Signal.html#method-c-list) to refer the number of SIGUSR2. Also, because `#termsig` already checks the status is signaled or not and may return `nil` if terminated, the call of `#signaled?` would be unnecessary.

```suggestion
        elsif status.termsig == Signal.list.fetch('USR2')
```

Or, it's even better to prepare a constant above like this:

```ruby
class Shell
  SIGUSR2 = Signal.list.fetch('USR2')
  ...

      elsif status.termsig == SIGUSR2
```

Reply:
Thanks for your kind review.

* modified the reference to signal number using a constants. also fixed test code. 
* remove the redundant `#signaled?` checking.